### ISSUE TITLE
Games give umu-proton error on launch

### ISSUE THREAD

**Issue by ZolaKluke** [lutris/lutris#5639]
### Bug description

Trying to launch any Windows game/application gives me an error
![image](https://github.com/user-attachments/assets/f7552e26-db2e-49d8-973a-08e2838128b6)

### How to Reproduce

1. Go to Lutris
2. Select game/application
3. Enable WINE/Proton
4. Get umu error

### Expected behavior

Applications to run fine like yesterday

### Log output

```shell
xander@fedora:~$ lutris -d
2024-10-07 02:36:28,815: Command 'fluidsynth' not found on your system
2024-10-07 02:36:29,053: Unable to run command, qdbus not found
INFO     2024-10-07 02:36:29,372 [application.do_command_line:479]:Starting Lutris 0.5.17
INFO     2024-10-07 02:36:29,612 [startup.run_all_checks:120]:NVIDIA GeForce RTX 3090 (10de:2204 1458:4042 nvidia) Driver 560.35.03
DEBUG    2024-10-07 02:36:29,763 [xrandr.get_resolutions:91]:Retrieving resolution list
DEBUG    2024-10-07 02:36:29,764 [xrandr._get_vidmodes:18]:Retrieving 65 video modes from XrandR
DEBUG    2024-10-07 02:36:29,933 [path_cache._update_missing_games:114]:Checking for missing games
DEBUG    2024-10-07 02:36:29,952 [api.check_stale_runtime_versions:52]:Modified at Mon 07 Oct 2024 01:05:13, will update after Mon 07 Oct 2024 07:05:13
DEBUG    2024-10-07 02:36:29,958 [lutriswindow.create_runtime_updater_cb:1244]:Runtime up to date
DEBUG    2024-10-07 02:36:31,191 [grid.on_item_activated:97]:Item activated: 1
WARNING  2024-10-07 02:36:31,242 [wine.set_regedit_keys:977]:Wine Virtual Desktop can't be used with Wine-GE and Proton
ERROR    2024-10-07 02:36:31,267 [exception_backstops.wrapper:36]:Plutonium Launcher (wine) has encountered an error: Install umu to use Proton
Traceback (most recent call last):
  File "/usr/lib/python3.12/site-packages/lutris/exception_backstops.py", line 31, in wrapper
    result = function(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/site-packages/lutris/game.py", line 668, in configure_game
    gameplay_info = self.get_gameplay_info(launch_ui_delegate)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/site-packages/lutris/game.py", line 607, in get_gameplay_info
    gameplay_info = self.runner.play()
                    ^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/site-packages/lutris/runners/wine.py", line 1216, in play
    command = self.get_command()
              ^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/site-packages/lutris/runners/wine.py", line 1188, in get_command
    raise MissingExecutableError("Install umu to use Proton")
lutris.exceptions.MissingExecutableError: Install umu to use Proton
DEBUG    2024-10-07 02:36:31,267 [game.stop_game:891]:Plutonium Launcher (wine) has run for 0 seconds
WARNING  2024-10-07 02:36:31,267 [game.stop_game:893]:The game has run for a very short time, did it crash?
DEBUG    2024-10-07 02:36:31,267 [application.on_game_stopped:794]:Removing 1 from running IDs
DEBUG    2024-10-07 02:36:31,274 [game.stop_game:903]:Playtime: 18 minutes
INFO     2024-10-07 02:36:33,705 [application.do_shutdown:1051]:Shutting down Lutris
xander@fedora:~$
```

### System Information

```shell
Vulkan support: YES
Esync support: YES
Fsync support: YES
Wine installed: YES
Gamescope: YES
Mangohud: YES
Gamemode: YES
Steam: YES
In Flatpak: NO
[System]
OS: Fedora Linux 40 KDE Plasma
Arch: x86_64
Kernel: 6.10.12-200.fc40.x86_64
Desktop: KDE
Display Server: wayland
[CPU]
Vendor: AuthenticAMD
Model: AMD Ryzen 9 7900X 12-Core Processor
Physical cores: 12
Logical cores: 24
[Memory]
RAM: 31.0 GB
Swap: 31.0 GB
[Graphics]
Vendor: NVIDIA Corporation
OpenGL Renderer: NVIDIA GeForce RTX 3090/PCIe/SSE2
OpenGL Version: 4.6.0 NVIDIA 560.35.03
OpenGL Core: 4.6.0 NVIDIA 560.35.03
OpenGL ES: OpenGL ES 3.2 NVIDIA 560.35.03
Vulkan Version: 1.3.290
Vulkan Drivers: NVIDIA GeForce RTX 3090 (1.3.280)
```

### Media (optional)

![image](https://github.com/user-attachments/assets/867ea298-8910-4a4f-87aa-3de86bb8ffcb)

**Comment by danieljohnson2** [comment]
I think this is due to changes in Umu; we've already got an update in the master, so 0.5.18 should fix this.

You can [try the master branch](https://github.com/lutris/lutris) and see if that will work. Download it, and run the ``bin/lutris`` script within. No need to install! Does that let you use Umu?

**Comment by ZolaKluke** [comment]
I'm not using umu though, I'm using GE-Proton9-13, it was working fine yesterday, I did run "sudo dnf update" yesterday and didn't reboot until this morning, did Lutris update on Fedora and break something?

**Comment by danieljohnson2** [comment]
Lutris updates its own runtime, and there was an update recently. That is what broke your build. This stuff is experimental (in 0.5.17 at least) and the experiment has gone a bit wrong. 0.5.18 will be better. Real soon now!

'GE-Proton9-13' is an Umu version. I find the naming here confusing, but i believe the idea is that we're calling Umu "Proton".

'GE-Proton9-13' is @GloriousEggroll's bleeding edge build of the 9.13 version of Wine, with Umu ("Proton").

To avoid Umu, use a 'wine-ge-' version. Those are pre-Umu.

**Comment by deus-vermiculus** [comment]
> Lutris updates its own runtime, and there was an update recently. That is what broke your build. This stuff is experimental (in 0.5.17 at least) and the experiment has gone a bit wrong. 0.5.18 will be better. Real soon now!
> 
> 'GE-Proton9-13' is an Umu version. I find the naming here confusing, but i believe the idea is that we're calling Umu "Proton".
> 
> 'GE-Proton9-13' is @GloriousEggroll's bleeding edge build of the 9.13 version of Wine, with Umu ("Proton").
> 
> To avoid Umu, use a 'wine-ge-' version. Those are pre-Umu.

yes it works. but since i am using the Flatpack version, i now have 2 Lutris installation on my machine and will have to messily remove one. I hope this will be pacthed out soon.

**Comment by danieljohnson2** [comment]
I hope we get 0.5.18 out pretty soon, but you should try a 'wine-ge-' version- you may not need Umu.

Thanks for running the test.

**Comment by cprin21** [comment]
Getting this same error on wine-ge-8-26. This broke Lutris for pre-umu proton versions.

**Comment by R1kaB3rN** [comment]
While this issue is closed as there's a solution for it, I'm going to chime in to clarify any confusion users have and why that exception is occurring.

Lutris recently pushed an update to its runtime where umu-launcher is being distributed as a single executable -- `umu-run`. However, in Lutris 0.5.17, the client searches for the file` umu_run.py` instead of `umu-run` in its runtime directory, hence the `MssingExecutableError`:

```
Traceback (most recent call last):
  File "/usr/lib/python3.12/site-packages/lutris/exception_backstops.py", line 31, in wrapper
    result = function(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/site-packages/lutris/game.py", line 668, in configure_game
    gameplay_info = self.get_gameplay_info(launch_ui_delegate)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/site-packages/lutris/game.py", line 607, in get_gameplay_info
    gameplay_info = self.runner.play()
                    ^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/site-packages/lutris/runners/wine.py", line 1216, in play
    command = self.get_command()
              ^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/site-packages/lutris/runners/wine.py", line 1188, in get_command
    raise MissingExecutableError("Install umu to use Proton")
lutris.exceptions.MissingExecutableError: Install umu to use Proton
```

To fix this until 0.5.18 gets released in your distribution, besides compiling the latest master branch from source, users need to make Lutris 0.5.17 find umu_run.py in its runtime directory. A simple way to do this is to rename umu-run to umu_run.py or create a symbolic link to umu-run in its runtime directory:

1. Open a terminal emulator
2. `cd $HOME/.local/share/lutris/runtime/umu`
3. `ln -s umu-run umu_run.py`

**Comment by cprin21** [comment]
Can confirm, this fixed the problem on 0.5.17 using the symlink as suggested by @R1kaB3rN .

**Comment by danieljohnson2** [comment]
This bug should not have affected the ordinary 'wine-ge-' runners. @cprin21, can you provide a log, or a screenshot of your game's Runner options (in the game configuration)? Something is not right.

**Comment by cprin21** [comment]
> can you provide a log, or a screenshot of your game's Runner options (in the game configuration)?
Sure
![Screenshot_20241006_123303](https://github.com/user-attachments/assets/b823123f-5632-47f1-b098-858d3e19a377)

**Comment by cprin21** [comment]
Also getting this after changing umu-launcher to umu_launcher.py or creating the symlink.

```
lutris-wrapper: Blizzard Battle.net
Started initial process 5535 from gamemoderun /home/admin/.local/share/lutris/runtime/umu/umu_run.py /home/admin/Games/battlenet/drive_c/Program Files (x86)/Battle.net/Battle.net Launcher.exe
Start monitoring process.
WARNING: '/home/admin/.local/share/umu/umu_util.py' is obsolete
WARNING: '/home/admin/.local/share/umu/reaper' is obsolete
WARNING: '/home/admin/.local/share/umu/umu_plugins.py' is obsolete
WARNING: '/home/admin/.local/share/umu/umu_dl_util.py' is obsolete
WARNING: '/home/admin/.local/share/umu/umu_run.py' is obsolete
WARNING: '/home/admin/.local/share/umu/umu_consts.py' is obsolete
WARNING: '/home/admin/.local/share/umu/umu_version.json' is obsolete
WARNING: '/home/admin/.local/share/umu/umu_log.py' is obsolete
WARNING: Runtime Platform not found
Restoring Runtime Platform...
Downloading latest steamrt sniper, please wait...
SteamLinuxRuntime_sniper.tar.xz: SHA256 is OK
WARNING: steamrt validation failed
WARNING: File does not exist: '/home/admin/.local/share/umu/pressure-vessel/bin/pv-verify'
/home/admin/.local/share/umu/run: line 22: /home/admin/.local/share/umu/pressure-vessel/bin/pressure-vessel-unruntime: No such file or directory
Monitored process exited.
Initial process has exited (return code: 32512)
All processes have quit
Exit with return code 32512
2024-10-06 12:36:07,096: Game still running (state: running)
2024-10-06 12:36:07,096: Stopping Blizzard Battle.net (wine)
```

**Comment by danieljohnson2** [comment]
This is very strange. I'm reading the code, and it runs this check before line 1188:
```python
def is_proton_path(wine_path) -> bool:
    return "Proton" in wine_path and "lutris" not in wine_path
```

To hit this exception, the Wine program it's running must have the word 'Proton' in the path. But it sure seems like it would not be. Very strange.

**Comment by cprin21** [comment]
This is a pretty big showstopping bug for Lutris as it currently stands. Running games as non-Steam games via Steam is a good workaround for now - for example this works for Battle.net, but this issue should probably be reopened for the sake of visibility.

**Comment by danieljohnson2** [comment]
Oh, it does not repro for me. The 'wine-ge-' versions work fine. I'm trying to figure out why it's broken for you- something strange is happening to you. But feel free to open a new issue for your problem if you wish- it's not the same issue and ought not to be consider closed.

The key thing is that you are getting Umu errors when not using Umu/Proton. That's different from @ZolaKluke's issue.

**Comment by danieljohnson2** [comment]
OK, @cprin21, I've got [a branch for you to try](https://github.com/lutris/lutris/tree/dj/0.5.17-log). Please download this- no need to install- and run the ``bin/lutris`` script, and run your game. It won't work- nothing is fixed- but the log output will contain a bunch of extra stuff. It should tell us what the path actually was. Post the spam and I'll have a look.

**Comment by cprin21** [comment]
> Please download this- no need to install- and run the `bin/lutris` script, and run your game. 

Unfortunately nothing occurs, no terminal output even, when bin/lutris is run. Also I discovered my issue is the same as ZolaKluke's. It took doing the umu-run workaround, manually setting my proton version to wine-ge-8-26, rolling back the workaround, and the game would run. 

With the umu-run to umu_run.py workaround enabled even wine-ge-8-26 would not run the game.

**Comment by danieljohnson2** [comment]
There is only one possible explanation, @cprin21. Your computer is haunted!

I might, alternately, guess you tried to run ``bin/lutris`` while Lutris was already running, and it just activated the existing window and exited.

**Comment by cprin21** [comment]
Got it (`bin/lutris -d`) to run after a reboot. Full console logs below. Tried running Battle Net, no success. (This only ran successfully once today) 

Without umu-run to umu_run.py workaround
```
INFO     2024-10-06 13:09:41,390 [application.do_command_line:479]:Starting Lutris 0.5.17
INFO     2024-10-06 13:09:41,502 [startup.run_all_checks:120]:AMD Radeon RX 6800 XT (1002:73bf 1da2:439e amdgpu) Driver 24.2.4
DEBUG    2024-10-06 13:09:41,766 [xrandr.get_resolutions:91]:Retrieving resolution list
DEBUG    2024-10-06 13:09:41,768 [xrandr._get_vidmodes:18]:Retrieving 30 video modes from XrandR
DEBUG    2024-10-06 13:09:41,806 [path_cache._update_missing_games:114]:Checking for missing games
DEBUG    2024-10-06 13:09:41,830 [api.check_stale_runtime_versions:52]:Modified at Sun 06 Oct 2024 11:56:12 AM, will update after Sun 06 Oct 2024 05:56:12 PM
DEBUG    2024-10-06 13:09:41,837 [lutriswindow.create_runtime_updater_cb:1244]:Runtime up to date
DEBUG    2024-10-06 13:09:43,225 [application.show_window:394]:Showing window EditGameConfigDialog1
DEBUG    2024-10-06 13:09:43,883 [xrandr.get_resolutions:91]:Retrieving resolution list
DEBUG    2024-10-06 13:09:43,885 [xrandr._get_vidmodes:18]:Retrieving 30 video modes from XrandR
DEBUG    2024-10-06 13:09:46,922 [game.save:450]:Saving Blizzard Battle.net (wine) with config ID blizzard-battlenet-standard-1726964379
DEBUG    2024-10-06 13:09:46,922 [config.save:227]:Saving LutrisConfig(level=game, game_config_id=blizzard-battlenet-standard-1726964379, runner=wine) config to /home/admin/.local/share/lutris/games/blizzard-battlenet-standard-1726964379.yml
DEBUG    2024-10-06 13:09:46,941 [path_cache.add_to_path_cache:47]:Adding Blizzard Battle.net (wine) to path cache
DEBUG    2024-10-06 13:09:46,949 [application.on_app_window_destroyed:419]:Removed window EditGameConfigDialog1
INFO     2024-10-06 13:09:48,894 [wine.get_executable:738]:+ version 1: None
INFO     2024-10-06 13:09:48,894 [wine.get_executable:741]:+ version 2: wine-ge-8-26-x86_64
INFO     2024-10-06 13:09:48,894 [wine.get_executable:745]:+ path 1: /home/admin/.local/share/lutris/runners/wine/wine-ge-8-26-x86_64/bin/wine
INFO     2024-10-06 13:09:48,896 [wine.get_executable:738]:+ version 1: None
INFO     2024-10-06 13:09:48,897 [wine.get_executable:741]:+ version 2: wine-ge-8-26-x86_64
INFO     2024-10-06 13:09:48,897 [wine.get_executable:745]:+ path 1: /home/admin/.local/share/lutris/runners/wine/wine-ge-8-26-x86_64/bin/wine
INFO     2024-10-06 13:09:48,897 [wine.get_executable:738]:+ version 1: None
INFO     2024-10-06 13:09:48,897 [wine.get_executable:741]:+ version 2: wine-ge-8-26-x86_64
INFO     2024-10-06 13:09:48,897 [wine.get_executable:745]:+ path 1: /home/admin/.local/share/lutris/runners/wine/wine-ge-8-26-x86_64/bin/wine
INFO     2024-10-06 13:09:48,897 [wine.get_command:1189]:Wine exe: /home/admin/.local/share/lutris/runners/wine/wine-ge-8-26-x86_64/bin/wine
INFO     2024-10-06 13:09:48,897 [wine.get_executable:738]:+ version 1: None
INFO     2024-10-06 13:09:48,897 [wine.get_executable:741]:+ version 2: wine-ge-8-26-x86_64
INFO     2024-10-06 13:09:48,897 [wine.get_executable:745]:+ path 1: /home/admin/.local/share/lutris/runners/wine/wine-ge-8-26-x86_64/bin/wine
INFO     2024-10-06 13:09:48,897 [wine.get_executable:738]:+ version 1: None
INFO     2024-10-06 13:09:48,897 [wine.get_executable:741]:+ version 2: wine-ge-8-26-x86_64
INFO     2024-10-06 13:09:48,897 [wine.get_executable:745]:+ path 1: /home/admin/.local/share/lutris/runners/wine/wine-ge-8-26-x86_64/bin/wine
INFO     2024-10-06 13:09:48,899 [wine.get_executable:738]:+ version 1: None
INFO     2024-10-06 13:09:48,899 [wine.get_executable:741]:+ version 2: wine-ge-8-26-x86_64
INFO     2024-10-06 13:09:48,899 [wine.get_executable:745]:+ path 1: /home/admin/.local/share/lutris/runners/wine/wine-ge-8-26-x86_64/bin/wine
DEBUG    2024-10-06 13:09:48,900 [xrandr.get_outputs:26]:Retrieving display outputs
DEBUG    2024-10-06 13:09:48,902 [xrandr._get_vidmodes:18]:Retrieving 30 video modes from XrandR
lutris-wrapper: Blizzard Battle.net
Started initial process 6399 from gamemoderun /home/admin/.local/share/lutris/runners/wine/wine-ge-8-26-x86_64/bin/wine /home/admin/Games/battlenet/drive_c/Program Files (x86)/Battle.net/Battle.net Launcher.exe
Start monitoring process.
gamemodeauto: 
fsync: up and running.
wine: RLIMIT_NICE is <= 20, unable to use setpriority safely
Initial process has exited (return code: 0)
Monitored process exited.
Exit with return code 0
DEBUG    2024-10-06 13:09:56,832 [monitored_command.on_stop:228]:Process 6397 has terminated with code 0
DEBUG    2024-10-06 13:09:56,920 [game.beat:924]:Game thread stopped
WARNING  2024-10-06 13:09:56,920 [game.on_game_quit:962]:Game still running (state: running)
INFO     2024-10-06 13:09:56,921 [game.stop:935]:Stopping Blizzard Battle.net (wine)
DEBUG    2024-10-06 13:09:56,921 [game.stop_game:891]:Blizzard Battle.net (wine) has run for 8 seconds
DEBUG    2024-10-06 13:09:56,922 [application.on_game_stopped:794]:Removing 1 from running IDs
DEBUG    2024-10-06 13:09:56,940 [game.stop_game:903]:Playtime: 2 hours 29 minutes
DEBUG    2024-10-06 13:09:56,940 [game.on_game_quit:980]:Blizzard Battle.net stopped at Sun, 06 Oct 2024 13:09:56
DEBUG    2024-10-06 13:09:56,956 [path_cache.add_to_path_cache:47]:Adding Blizzard Battle.net (wine) to path cache
```

With umu-run to umu_run.py workaround
```
INFO     2024-10-06 13:12:40,412 [application.do_command_line:479]:Starting Lutris 0.5.17
INFO     2024-10-06 13:12:40,523 [startup.run_all_checks:120]:AMD Radeon RX 6800 XT (1002:73bf 1da2:439e amdgpu) Driver 24.2.4
DEBUG    2024-10-06 13:12:40,764 [xrandr.get_resolutions:91]:Retrieving resolution list
DEBUG    2024-10-06 13:12:40,765 [xrandr._get_vidmodes:18]:Retrieving 30 video modes from XrandR
DEBUG    2024-10-06 13:12:40,802 [path_cache._update_missing_games:114]:Checking for missing games
DEBUG    2024-10-06 13:12:40,825 [api.check_stale_runtime_versions:52]:Modified at Sun 06 Oct 2024 11:56:12 AM, will update after Sun 06 Oct 2024 05:56:12 PM
DEBUG    2024-10-06 13:12:40,834 [lutriswindow.create_runtime_updater_cb:1244]:Runtime up to date
DEBUG    2024-10-06 13:12:45,335 [application.show_window:394]:Showing window EditGameConfigDialog1
DEBUG    2024-10-06 13:12:45,958 [xrandr.get_resolutions:91]:Retrieving resolution list
DEBUG    2024-10-06 13:12:45,960 [xrandr._get_vidmodes:18]:Retrieving 30 video modes from XrandR
DEBUG    2024-10-06 13:12:48,332 [game.save:450]:Saving Blizzard Battle.net (wine) with config ID blizzard-battlenet-standard-1726964379
DEBUG    2024-10-06 13:12:48,332 [config.save:227]:Saving LutrisConfig(level=game, game_config_id=blizzard-battlenet-standard-1726964379, runner=wine) config to /home/admin/.local/share/lutris/games/blizzard-battlenet-standard-1726964379.yml
DEBUG    2024-10-06 13:12:48,353 [path_cache.add_to_path_cache:47]:Adding Blizzard Battle.net (wine) to path cache
DEBUG    2024-10-06 13:12:48,361 [application.on_app_window_destroyed:419]:Removed window EditGameConfigDialog1
INFO     2024-10-06 13:12:49,203 [wine.get_executable:738]:+ version 1: None
INFO     2024-10-06 13:12:49,203 [wine.get_executable:741]:+ version 2: wine-ge-8-26-x86_64
INFO     2024-10-06 13:12:49,203 [wine.get_executable:745]:+ path 1: /home/admin/.local/share/lutris/runners/wine/wine-ge-8-26-x86_64/bin/wine
INFO     2024-10-06 13:12:49,205 [wine.get_executable:738]:+ version 1: None
INFO     2024-10-06 13:12:49,205 [wine.get_executable:741]:+ version 2: wine-ge-8-26-x86_64
INFO     2024-10-06 13:12:49,205 [wine.get_executable:745]:+ path 1: /home/admin/.local/share/lutris/runners/wine/wine-ge-8-26-x86_64/bin/wine
INFO     2024-10-06 13:12:49,206 [wine.get_executable:738]:+ version 1: None
INFO     2024-10-06 13:12:49,206 [wine.get_executable:741]:+ version 2: wine-ge-8-26-x86_64
INFO     2024-10-06 13:12:49,206 [wine.get_executable:745]:+ path 1: /home/admin/.local/share/lutris/runners/wine/wine-ge-8-26-x86_64/bin/wine
INFO     2024-10-06 13:12:49,206 [wine.get_command:1189]:Wine exe: /home/admin/.local/share/lutris/runners/wine/wine-ge-8-26-x86_64/bin/wine
INFO     2024-10-06 13:12:49,206 [wine.get_executable:738]:+ version 1: None
INFO     2024-10-06 13:12:49,206 [wine.get_executable:741]:+ version 2: wine-ge-8-26-x86_64
INFO     2024-10-06 13:12:49,206 [wine.get_executable:745]:+ path 1: /home/admin/.local/share/lutris/runners/wine/wine-ge-8-26-x86_64/bin/wine
INFO     2024-10-06 13:12:49,206 [wine.get_executable:738]:+ version 1: None
INFO     2024-10-06 13:12:49,206 [wine.get_executable:741]:+ version 2: wine-ge-8-26-x86_64
INFO     2024-10-06 13:12:49,206 [wine.get_executable:745]:+ path 1: /home/admin/.local/share/lutris/runners/wine/wine-ge-8-26-x86_64/bin/wine
INFO     2024-10-06 13:12:49,208 [wine.get_executable:738]:+ version 1: None
INFO     2024-10-06 13:12:49,208 [wine.get_executable:741]:+ version 2: wine-ge-8-26-x86_64
INFO     2024-10-06 13:12:49,208 [wine.get_executable:745]:+ path 1: /home/admin/.local/share/lutris/runners/wine/wine-ge-8-26-x86_64/bin/wine
DEBUG    2024-10-06 13:12:49,208 [xrandr.get_outputs:26]:Retrieving display outputs
DEBUG    2024-10-06 13:12:49,210 [xrandr._get_vidmodes:18]:Retrieving 30 video modes from XrandR
lutris-wrapper: Blizzard Battle.net
Started initial process 7023 from gamemoderun /home/admin/.local/share/lutris/runners/wine/wine-ge-8-26-x86_64/bin/wine /home/admin/Games/battlenet/drive_c/Program Files (x86)/Battle.net/Battle.net Launcher.exe
Start monitoring process.
gamemodeauto: 
fsync: up and running.
wine: RLIMIT_NICE is <= 20, unable to use setpriority safely
Initial process has exited (return code: 0)
Monitored process exited.
Exit with return code 0
DEBUG    2024-10-06 13:12:56,939 [monitored_command.on_stop:228]:Process 7021 has terminated with code 0
DEBUG    2024-10-06 13:12:57,228 [game.beat:924]:Game thread stopped
WARNING  2024-10-06 13:12:57,228 [game.on_game_quit:962]:Game still running (state: running)
INFO     2024-10-06 13:12:57,228 [game.stop:935]:Stopping Blizzard Battle.net (wine)
DEBUG    2024-10-06 13:12:57,228 [game.stop_game:891]:Blizzard Battle.net (wine) has run for 8 seconds
DEBUG    2024-10-06 13:12:57,228 [application.on_game_stopped:794]:Removing 1 from running IDs
DEBUG    2024-10-06 13:12:57,237 [game.stop_game:903]:Playtime: 2 hours 29 minutes
DEBUG    2024-10-06 13:12:57,237 [game.on_game_quit:980]:Blizzard Battle.net stopped at Sun, 06 Oct 2024 13:12:57
DEBUG    2024-10-06 13:12:57,246 [path_cache.add_to_path_cache:47]:Adding Blizzard Battle.net (wine) to path cache
```
```

**Comment by danieljohnson2** [comment]
Well, that log looks just fine, no problems- and no trace-back. It looks like the game ran just fine- using ``wine-ge-8-26``.

Can you no longer reproduce the failure now that you have fiddled with Umu?

**Comment by cprin21** [comment]
Very odd, perhaps it is an issue Battle Net has with wine-ge-8-26 and not umu. The failure as in the popup claiming umu is not installed does indeed not show up.

**Comment by danieljohnson2** [comment]
Well, I see something in this new log. It has this wine: ``/home/admin/.local/share/Steam/steamapps/common/Proton - Experimental/files/bin/wine``. That has the magic word Proton in it, which would trigger the Umu logic (it's a bit janky)

Does this game work with the master branch code?

**Comment by strycore** [comment]
> 'GE-Proton9-13' is an Umu version. I find the naming here confusing, but i believe the idea is that we're calling Umu "Proton".

Umu is not Proton. Umu is not Wine. Umu is not a Lutris runner. Umu is a software layer that enables you to use Proton in Lutris or other launchers.

This stuff should be internal only, users should not know about the existence of umu or what it's supposed to do.

**Comment by danieljohnson2** [comment]
Well, I don't think I'm the only one who is confused about it!

**Comment by strycore** [comment]
That's why the existence of umu should be known only by people who can understand it.

**Comment by danieljohnson2** [comment]
Apparently that's not me... :frowning_face:

**Comment by cprin21** [comment]
> Well, I see something in this new log. It has this wine: `/home/admin/.local/share/Steam/steamapps/common/Proton - Experimental/files/bin/wine`. That has the magic word Proton in it, which would trigger the Umu logic (it's a bit janky)

Sorry but no, Proton Experimental is only available with the umu_run.py workaround but then launching using any version of Proton made available by that workaround causes that console output  regardless of the master branch code or not. 
Proton/Wine versions before umu was implemented, e.g. wine-ge-8-26 do not show the warnings that such things are obsolete and do launch games. It launched GOG Galaxy via wine-ge-8-26 for example so there must be something borked with my Battle.net install.

**Comment by strycore** [comment]
> Apparently that's not me... ☹️

I believe you have enough background to get a solid grasp of what umu is doing.  I was mostly referring to people on Reddit or news websites who have consistently shown their inability to comprehend what umu is (and are not  even close to getting it).

This just means, that at no point in the Lutris UI should we show the name "umu", this should be kept internal and the only visible name should be Proton. The use of Proton implies the use of umu and vice versa.

**Comment by danieljohnson2** [comment]
Well, @cprin21, I don't really understand. I suspect things are going wrong when you run a game that has 'Proton' in the path, and that may explain why GOG Galaxy still works. I would have thought Umu would be just the thing to use with 'Proton - experimental' stuff, but I guess not.

You might try moving (or copying) a problem game into ``~/Games`` to see if that changes anything.

**Comment by cprin21** [comment]
Master branch now does resolve this problem of post-umu proton versions working. This problem should not have ever happened to the stable build of Lutris.

**Comment by danieljohnson2** [comment]
Well, Umu in Lutris 0.5.17 was always experimental. It's not a shock that it's been a bit broken. The main thing is that it shouldn't affect 'normal' use. I'm afraid that wasn't really true, and I'm sorry for that.

I am inclined to think that Umu/Proton logic should be activated _exclusively_ by the Wine version you select, never the path to the wine binary.

**Comment by danieljohnson2** [comment]
I've put together a PR (#5641) to remove the logic where Lutris searches the wine path for 'Proton'. Should make this sort of thing more controllable. I hope it will help with this problem.

**Comment by ACiDxCHRiST** [comment]
> To fix this until 0.5.18 gets released in your distribution, besides compiling the latest master branch from source, users need to make Lutris 0.5.17 find umu_run.py in its runtime directory. A simple way to do this is to rename umu-run to umu_run.py or create a symbolic link to umu-run in its runtime directory:
> 
>     1. Open a terminal emulator
> 
>     2. `cd $HOME/.local/share/lutris/runtime/umu`
> 
>     3. `ln -s umu-run umu_run.py`

Do you know an equivalent fix for Lutris flatpak? I do not have a `~/.local/share/lutris` folder

**Comment by ACiDxCHRiST** [comment]
> ```
> cd ~/.var/app/net.lutris.Lutris/data/lutris/runtime/umu
> ln -s umu-run umu_run.py
> ```

Ok all the protons I have installed showed back up in the Wine Version drop down menu but now nothing runs if I choose one of them. The Lutris item just says "Launching" indefinitely. When I check the wine prefix folder that I chose, there are only a couple folders total. It doesn't look like the initial prefix setup is completing. I can see it created `C:\users\deck` and about nothing else.

Does anyone know what the problem is or is there a log file somewhere that would have helpful information?
